gslib/commands/signurl.py

```diff
@@ -48,23 +48,33 @@
 from gslib.utils import constants
 from gslib.utils.boto_util import GetNewHttp
 from gslib.utils.shim_util import GcloudStorageMap, GcloudStorageFlag
-from gslib.utils.signurl_helper import CreatePayload, GetFinalUrl
+from gslib.utils.signurl_helper import CreatePayload, GetFinalUrl, to_bytes
 
 try:
   # Check for openssl.
   # pylint: disable=C6204
   from OpenSSL.crypto import FILETYPE_PEM
-  from OpenSSL.crypto import load_pkcs12
   from OpenSSL.crypto import load_privatekey
   from OpenSSL.crypto import sign
+  from OpenSSL.crypto import PKey
   HAVE_OPENSSL = True
 except ImportError:
   load_privatekey = None
-  load_pkcs12 = None
   sign = None
   HAVE_OPENSSL = False
   FILETYPE_PEM = None
 
+try:
+  from cryptography.hazmat.primitives import hashes
```

Comment:
We are removing dependency on pyopenssl and adding dependency on cryptography? Is cryptograpgy included as dependency?

Reply:
We're not removing the dependency of openSSL, we're removing the dependency on load_pkcs12 attribute of openSSL as it's deprecated. We're using cryptography module to decode to .p12 keys.